In [5]:
from sklearn.svm import SVC
from sklearn.ensemble import RandomForestClassifier
from sklearn.linear_model import LogisticRegression
from sklearn.naive_bayes import GaussianNB
from sklearn.naive_bayes import MultinomialNB
from sklearn.tree import DecisionTreeClassifier

import pandas as pd
import numpy as np
from sklearn.datasets import load_digits
from sklearn.model_selection import GridSearchCV
import warnings
warnings.filterwarnings("ignore")

digits = load_digits()

### Models with different hyperparameters

In [11]:
model_dict = {
    'svm': {
        'model': SVC(gamma='auto'),
        'params': {
            'C': [1,10,20],
            'kernel': ['rbf','linear']
        }
    },
    'random_forest': {
        'model': RandomForestClassifier(),
        'params': {
            'n_estimators': [1,5,10]
        }
    },
    'logistic_regression': {
        'model': LogisticRegression(solver='liblinear',multi_class='auto'),
        'params': {
            'C': [1,5,10]
        }
    },
    'NaiveBayes_Gaussian': {
        'model': GaussianNB(),
        'params': {
            'var_smoothing': [1e-09, 0.5, 1]
        }
    },
    'NaiveBayes_Multinomial': {
        'model': MultinomialNB(),
        'params': {
            'alpha': [1,5,10]
        }
    },
    'Decision_Tree': {
        'model': DecisionTreeClassifier(),
        'params': {
            'criterion': ['gini','entropy']
        }
    }
}

### GridSearchCV

In [12]:
scores = []

for model_name, mp in model_dict.items():
    clf = GridSearchCV(mp['model'],mp['params'], cv=5, return_train_score=False)
    clf.fit(digits.data,digits.target)
    scores.append({
        'model': model_name,
        'best_score': clf.best_score_,
        'params': clf.best_params_
    })

df = pd.DataFrame(scores, columns=['model','params','best_score'])
df

model                        params  best_score
0                     svm  {'C': 1, 'kernel': 'linear'}    0.947697
1           random_forest          {'n_estimators': 10}    0.906530
2     logistic_regression                      {'C': 1}    0.922114
3     NaiveBayes_Gaussian        {'var_smoothing': 0.5}    0.885376
4  NaiveBayes_Multinomial                 {'alpha': 10}    0.874246
5           Decision_Tree      {'criterion': 'entropy'}    0.803036

**Support Vector Machine has the best score out of all the models**
-  **parameters**: {'C': 1, 'kernel': 'linear'}
-  **best score**: 0.947697
-  **accuracy**: 94.76%